src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2367,8 +2367,16 @@ protected static bool TypesEquivalent(ITypeSymbol? oldType, ITypeSymbol? newType
         protected static bool TypesEquivalent<T>(ImmutableArray<T> oldTypes, ImmutableArray<T> newTypes, bool exact) where T : ITypeSymbol
             => oldTypes.SequenceEqual(newTypes, exact, (x, y, exact) => TypesEquivalent(x, y, exact));
 
-        protected static bool ParameterTypesEquivalent(IParameterSymbol oldParameter, IParameterSymbol newParameter, bool exact)
-            => (exact ? s_exactSymbolEqualityComparer : s_runtimeSymbolEqualityComparer).ParameterEquivalenceComparer.Equals(oldParameter, newParameter);
+        protected static bool ParameterTypesEquivalent(IParameterSymbol oldParameter, IParameterSymbol newParameter, bool exact, bool allowDiscards)
+        {
+            if (exact)
+            {
+                return s_exactSymbolEqualityComparer.ParameterEquivalenceComparer.Equals(oldParameter, newParameter);
+            }
+
+            return s_runtimeSymbolEqualityComparer.ParameterEquivalenceComparer.Equals(oldParameter, newParameter) ||
+                (allowDiscards && (oldParameter.IsDiscard || newParameter.IsDiscard));
```

Comment:
>(oldParameter.IsDiscard || newParameter.IsDiscard [](http://example.com/codeflow?start=34&length=49)

Shouldn't discards be handled in the ParameterEquivalenceComparer itself? Why are discards handled differently then name change?

Reply:
I thought about that, and it seemed odd. For our purposes a discard is equal to any and every other parameter, but it seems like that would break a _lot_ of things.

Reply:
> discard is equal to any and every other parameter

I don't think that's true. Discard should be only equal to the other corresponding parameter if they have the same type, but they can differ in names. This should imo be implemented in ParameterEquivalenceComparer, where name is compared.